openbas-api/src/main/java/io/openbas/rest/injector_contract/InjectorContractApi.java

```diff
@@ -61,17 +63,17 @@ public Page<InjectorContract> injectorContracts(@RequestBody @Valid final Search
                     Filters.FilterGroup newFilterGroup = new Filters.FilterGroup();
                     newFilterGroup.setFilters(searchPaginationInput.getFilterGroup().getFilters().stream().filter(filter -> !filter.getKey().equals("injector_contract_kill_chain_phases")).toList());
                     newSearchPaginationInput.setFilterGroup(newFilterGroup);
-                    return buildPaginationJPA(
-                            (Specification<InjectorContract> specification, Pageable pageable) -> this.injectorContractRepository.findAll(
-                                    InjectorContractSpecification.fromKillChainPhase(killChainPhaseFilter.getValues().getFirst()).and(specification), pageable),
-                            newSearchPaginationInput,
-                            InjectorContract.class
+                    return buildPaginationCriteriaBuilder(
+                        (Specification<InjectorContract> specification, Pageable pageable) -> this.injectorContractService.injectorContracts(
+                            InjectorContractSpecification.fromKillChainPhase(killChainPhaseFilter.getValues().getFirst()).and(specification), pageable),
+                        searchPaginationInput,
```

Comment:
Why switching ? If that is intentional, we should remove newSearchPaginationInput as it's not used anymore.